draft-ietf-jsonpath-base.md

```diff
@@ -1078,9 +1080,10 @@ Notes:
 <!-- issue: comparison with structured value -->
 * Types are not implicitly converted in comparisons.
   So `"13 == '13'"` selects no node.
-* A member or element value by itself is *falsy* only, if it does not exist. Otherwise it is *truthy*, resulting in its value. To be more specific explicit comparisons are necessary. This existence test — as an exception of the general rule — also works with complex values.
+* A member or element value by itself is *falsy* only, if it does not exist. Otherwise it is *truthy*, resulting in its value. To be more specific explicit comparisons are necessary. This existence test — as an exception of the general rule — also works with structured values.
 * Regular expression tests can be applied to `string` values only.
-* Containment tests work with arrays and objects.
+* The value of LHS `containable` of a `contain-expr` is compared to every single element of the RHS `container`. In case of a match a selection occurs. Containment tests — like comparisons — are restricted to primitive values. So even if a structured `containable` value is equal to a certain structured value in `container`, no selection is done.
```

Comment:
Why this restriction? This behaviour would probably be surprising to most users.

I think there was resistance to implementing deep equals, but I'm still not sure there's much of an issue in doing that.

Reply:
Motivation here was consistency with `comp-expr`, which is restricted to primitives. So loosening this restriction should also impact comparisons. Do we really want to open this Pandora's Box?

Reply:
> Motivation here was consistency with comp-expr, which is restricted to primitives. So loosening this restriction should also impact comparisons. Do we really want to open this Pandora's Box?

Perhaps; perhaps not. But I'd like to understand what issues there are, if any, specifying the behaviour of equality tests of non-primitives (essentially, deep equals). I think we agreed at the last meeting that _implementing_ deep equals is relatively straightforward.